### PR TITLE
feat: improve memtable conflicts error message

### PR DESCRIPTION
**Pull request by fuyufjh**
I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## What's changed and what's your intention?

Make the panic message easier to read when conflicts detected by memtable.

Before:

> thread 'risingwave-streaming-actor' panicked at 'invalid flush status: double insert [75, 0, 128, 0, 0, 4, 0, 128, 0, 0, 4] -> [1, 4, 0, 0, 0, 1, 18, 0, 0, 0, 83, 117, 112, 112, 108, 105, 101, 114, 35, 48, 48, 48, 48, 48, 48, 48, 48, 52, 1, 25, 0, 0, 0, 66, 107, 55, 97, 104, 52, 67, 75, 56, 83, 89, 81, 84, 101, 112, 69, 109, 118, 77, 107, 107, 103, 77, 119, 103, 1, 15, 0, 0, 0, 50, 53, 45, 56, 52, 51, 45, 55, 56, 55, 45, 55, 52, 55, 57, 1, 0, 0, 4, 0, 98, 25, 56, 25, 2, 0, 0, 0, 0, 0, 0, 0, 1, 4, 0, 0, 0]'

After:

> thread 'table::streaming_table::test_streaming_table::test_mem_table_assertion' panicked at 'mem-table operation conflicts! key: OrderedRow([NormalOrder(Some(Int32(-2139095040)))]), prev: Insert(Row([Some(Int32(1)), Some(Int32(11)), Some(Int32(111))])), new: Insert(Row([Some(Int32(1)), Some(Int32(22)), Some(Int32(222))]))'

## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Documentation

not user-facing

## Refer to a related PR or issue link (optional)

None

### CODE REVIEW

File: src/storage/src/table/streaming_table/state_table.rs [hunk 1]
```diff
@@ -359,6 +378,38 @@ impl<S: StateStore> StateTable<S> {
 
 // write
 impl<S: StateStore> StateTable<S> {
+    fn pretty_row_op(row_op: &RowOp, data_types: &[DataType]) -> String {
```
Comment (BugenZhao) [inline src/storage/src/table/streaming_table/state_table.rs:381]: How about directly implementing this method on `RowOp` itself?

**Review by BugenZhao — APPROVED** [review]
LGTM

File: src/storage/src/table/streaming_table/state_table.rs (hunk 1 above)
Comment (fuyufjh) [reply src/storage/src/table/streaming_table/state_table.rs:381]: It needs `data_types` to be passed, as a result, the function still looks ugly 🥲

File: src/storage/src/table/streaming_table/state_table.rs [hunk 2]
```diff
@@ -59,6 +60,9 @@ pub struct StateTable<S: StateStore> {
     /// Used for serializing the primary key.
     pk_serializer: OrderedRowSerializer,
 
+    /// Used for deserializing the primary key. Debug-only for now.
+    pk_deserializer: OrderedRowDeserializer,
```
Comment (fuyufjh) [inline src/storage/src/table/streaming_table/state_table.rs:64]: I think `OrderedRowSerializer` and `OrderedRowDeserializer` should be merged into one later.

Currently `OrderedRowSerializer` doesn't store data types inside, but I believe this is not correct.

**Pull request merged**